src/atomic_structure.hh

```diff
@@ -122,12 +122,16 @@ namespace rascal {
      */
     void wrap() {
       auto scaled_positions = this->get_scaled_positions();
-      auto functor{math::MakePositivePyMod(1.)};
 
       for (int i_dim{0}; i_dim < Dim; ++i_dim) {
         if (this->pbc[i_dim]) {
           scaled_positions.row(i_dim) =
-              scaled_positions.row(i_dim).unaryExpr(functor);
+              scaled_positions.row(i_dim).unaryExpr([](double d) {
+                // Modulo that follows python standard, i.e. (-3) % 5 == 2
+                // (python) and not (-3) % 5 == -3 (C++).
+                auto m = std::fmod(d, 1.0);
+                return m + (m < 0 ? 1.0 : 0);
+              });
```

Comment:
> This operation is only used once now but we might need it

Then a new functor should be created if there is really a need to abstract this particular operation. But moving the implementation close to the place of use mean there is less jumping around file to understand a particular piece of code.

> Also it is the kind of mind blow (difference in behavior for the mod between cpp and python) that I would like to keep very explicit, like having a dedicated function.

I think having the comment inline is the more explicit this can be.